crates/polars-plan/src/logical_plan/optimizer/slice_pushdown_lp.rs

```diff
@@ -324,7 +324,7 @@ impl SlicePushDown {
             (Projection {input, expr, schema, options}, Some(_)) => {
                 // The slice operation may only pass on simple projections. col("foo").alias("bar")
                 if expr.iter().all(|root|  {
-                    aexpr_is_elementwise(*root, expr_arena)
+                    aexpr_is_elementwise_and_has_column(*root, expr_arena)
```

Comment:
I think we can have scalar literals if there is at least one normal column projection. So I think we must relax this check a little bit.

```python
query
.select(
 col("foo"), lit(1)
)
```

can benefit from slice pushdown.